src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -22,34 +22,30 @@ CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 source "${CURRENT_DIR}/../integration_test_setup.sh" \
   || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
 
+# Check if all the tools required by CC coverage are installed.
+[[ ! -x /usr/bin/lcov ]] && fail "lcov not installed. Skipping test."
+[[ -z $( which gcov ) ]] && fail "gcov not installed."
+[[ -z $( which g++ ) ]] && fail "g++ not installed."
+
+# These are the variables needed by tools/test/collect_cc_coverage.sh
+# They will be properly sub-shelled when invoking the script.
+readonly coverage_dir="${PWD}"
+readonly coverage_gcov_path="${PWD}/mygcov"
+readonly root="${PWD}"
+readonly coverage_manifest="${PWD}/coverage_manifest.txt"
+readonly coverage_output_file="${PWD}/coverage_report.dat"
+
+# Setup to be run for every test.
 function set_up() {
-  if [[ ! -x /usr/bin/lcov ]]; then
-    fail "lcov not installed. Skipping test."
-  fi
-
-  if [[ -z $( which gcov ) ]]; then
-    fail "gcov not installed."
-  fi
-
-  if [[ -z $( which g++ ) ]]; then
-    fail "g++ not installed."
-  fi
-
   # The script expects gcov to be at $COVERAGE_GCOV_PATH.
-  gcov_location=$( which gcov )
-  cp $gcov_location "${PWD}/mygcov"
+  local gcov_location=$( which gcov )
+  cp "$gcov_location" "${PWD}/mygcov"
```

Comment:
You can just do:
cp "(which gcov)" "${PWD}/mygcov"

Reply:
Actually:
cp "(which gcov)" "$COVERAGE_GCOV_PATH"

would be better.

Reply:
Done